design/one-pager-ignore-changes.md

```diff
@@ -0,0 +1,157 @@
+provider maintainers could just use this function in the Update method, without the need to invent something themselves, of
+course unless its necessary.
+
+TODO: insert some pseudocode or even real code and use it in a provider update
+
+#### Feature Gating
+
+Similar to all other new features being added to Crossplane, we will ship this
+new policy as an alpha feature that will be off by default and will be
+controlled by `--enable-alpha-ignore-changes` flag in Providers.
+
+This will not prevent the field from appearing in the schema of the managed
+resources. However, we will ignore the `spec.ignoreChanges` when the feature
+is not enabled.
+
+### Implementation alternative considered
+
+An alternative was considered where the above helper function which unsets fields in the managed resource would be used
+directly in the [crossplane-runtime] reconciler, and then the a version of the managed resource without the ignored fields
+would be passed to the Update method.
+
+While this solution would enable the functionality without the need to change all the providers, it has a few 
+drawbacks:
+* It doesn't allow for flexibility or handling special cases in the provider code, as the providers will just receive a 
+managed resource which does not have the ignored fields set. There could be cases where the wish is to not update the field
+on the external resource, but is useful in some other way.
+* Won't work that good with Upjet as it uses Terraform underneath which would work better with its native [ignore_changes].
+
+## Solution alternative considered
+
+The idea here was to do something similar to K8s `initContainers` and add a field `initProvider` to the managed resources.
+The parameters under `initProvider` would be used only during creation.
+
+```yaml
+apiVersion: eks.aws.crossplane.io/v1alpha1
+kind: NodeGroup
+metadata:
+  name: my-group
+spec:
+  initProvider:
+    scalingConfig:
+      desiredSize: 1
+  forProvider:
+    region: us-east-1
+    clusterNameRef:
+      name: sample-cluster
+    subnetRefs:
+      - name: sample-subnet1
+    nodeRoleRef:
+      name: somenoderole
+    scalingConfig:
+      maxSize: 5
+      minSize: 1
+    updateConfig:
+      maxUnavailablePercentage: 50
+      force: true
+  providerConfigRef:
+    name: example
+```
+
+The great thing about this solution is that it feels natural to add `initProvider` as we already have `forProvider`,
+`atProvider`. But on the other hand, it loses some flexibilty when it's combined with *late initialization* as the 
+when the `forProvider` field gets late initialized, we would need additional logic to have that field ignored. Which 
+brings us back to the proposed solution above. Of course, we could also not *late initialize* fields that are
+in `initProvider`.
+
+Another thing that would be bothersome with this solution is that a value is needed in the `initProvider`. A field
+cannot just be set to be ignored, without worrying about the initial value.
```

Comment:
Do we have a some real life examples where this would be an issues?

Reply:
Note to myself :) 

I think no matter which API we want to proceed with, we would use some parameters during `creation` but ignore them during `updates`. However, the main difference here is the user cannot use the default (late-inited) value if they want that parameter to be ignored and has to provide a value to be used during creation.

Reply:
Yeah, thats the main point. I have been trying to find some example of this but TBH I dont have the experience to know where things like this can happpen.